rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/AbstractUndoLogManager.java

```diff
@@ -339,20 +339,22 @@ public void undo(DataSourceProxy dataSourceProxy, String xid, long branchId) thr
                             State.GlobalFinished.name());
                     }
                 } else {
-                    insertUndoLogWithGlobalFinished(xid, branchId, UndoLogParserFactory.getInstance(), conn);
-                    conn.commit();
-                    if (LOGGER.isInfoEnabled()) {
-                        LOGGER.info("xid {} branch {}, undo_log added with {}", xid, branchId,
-                            State.GlobalFinished.name());
+                    try {
+                        insertUndoLogWithGlobalFinished(xid, branchId, UndoLogParserFactory.getInstance(), conn);
```

Comment:
Why is it modified here?

Reply:
> The changes here are to narrow the scope of the catch for SQLIntegrityConstraintViolationException at the outermost level, only handling conflicts with the unique index on the undo_log. If the unique index conflict also occurs when executing the rollback undoExecutor.executeOn for the business, it will enter a deadlock directly here.

I am very sorry, I replied to this paragraph one day after you replied, but it has been pending. I didn't know I had to click submitted. disturb you

Reply:
> The changes here are to narrow the scope of the catch for SQLIntegrityConstraintViolationException at the outermost level, only handling conflicts with the unique index on the undo_log. If the unique index conflict also occurs when executing the rollback undoExecutor.executeOn for the business, it will enter a deadlock directly here.

Reply:
> The changes here are to narrow the scope of the catch for SQLIntegrityConstraintViolationException at the outermost level, only handling conflicts with the unique index on the undo_log. If the unique index conflict also occurs when executing the rollback undoExecutor.executeOn for the business, it will enter a deadlock directly here.

<img width="808" alt="Snipaste_2024-07-09_19-43-48" src="https://github.com/apache/incubator-seata/assets/48412175/5ffb6bde-e3ab-45b2-94c8-97b197c319d4">

Reply:
> > The changes here are to narrow the scope of the catch for SQLIntegrityConstraintViolationException at the outermost level, only handling conflicts with the unique index on the undo_log. If the unique index conflict also occurs when executing the rollback undoExecutor.executeOn for the business, it will enter a deadlock directly here.
> 
> <img alt="Snipaste_2024-07-09_19-43-48" width="808" src="https://private-user-images.githubusercontent.com/48412175/346945446-5ffb6bde-e3ab-45b2-94c8-97b197c319d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2OTI4MzUsIm5iZiI6MTcyMjY5MjUzNSwicGF0aCI6Ii80ODQxMjE3NS8zNDY5NDU0NDYtNWZmYjZiZGUtZTNhYi00NWIyLTk0YzgtOTdiMTk3YzMxOWQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDEzNDIxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzYjA1MTM2NmQyNzFlNzkyMGRkNjUwZWNkYjY0Mzc0MmJiZTkxNjI5N2FmNmUxM2FlMjExMTAxNzg2MjhmZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GDWMSPrYkc0w7rnt4pN5SD84NXu1AYKsP6SPHqWPfwQ">

Sorry, I just saw your message now. I'll review the code as soon as possible.